skeleton/gitops-template/components/http/base/rhoai/initialize-dsp.yaml

```diff
@@ -90,6 +90,13 @@ spec:
                       successThreshold: 1
                       timeoutSeconds: 1
                     env:
+                      {%- if values.includeModelEndpointSecret %}
+                      - name: MODEL_ENDPOINT_BEARER
+                        valueFrom:
+                          secretKeyRef:
+                            name: $${{   values.modelEndpointSecretName   }}
```

Comment:
Application pod:
```
env:
        - name: MODEL_ENDPOINT_BEARER
          valueFrom:
            secretKeyRef:
              name: ibm-granite-8b-app
              key: api_key
```

Notebook Pod:
```
env:
            - name: MODEL_ENDPOINT_BEARER
              valueFrom:
                secretKeyRef:
                  name: '-granite-8b-app'
                  key: ''
```
and this is the error on RHOAI UI:
<img width="534" alt="Screenshot 2024-10-17 at 3 16 35 PM" src="https://github.com/user-attachments/assets/4258eb45-738c-4e06-b535-b01cb1d93b59">

I confirmed that I entered the secret and key correctly:
<img width="575" alt="Screenshot 2024-10-17 at 3 14 12 PM" src="https://github.com/user-attachments/assets/c9dce446-03c4-4141-bda5-bcdfd383a64d">

The secret name is getting truncated? Also, the key is empty! 🤔

Reply:
Is it because of the double `$$` and the extra whitespaces within `{{ }}`?

Reply:
Hmm, good catch.. I'm trying to reproduce it too because I didn't see that in the past (might be a customization made from my side).

I've added a fix here: https://github.com/redhat-ai-dev/ai-lab-app/pull/33 so the `modelEndpointSecretName` and `modelEndpointSecretKey` follow the same format with all the other values we use in the `initialize-dsp`.

Reply:
I've added a `rag4` in my fork's main branch with this change and for me it works with this update.

* You can add the secret in a namespace of your preference.
* Import https://github.com/thepetk/ai-lab-template/blob/main/templates/rag4/template.yaml

![Screenshot from 2024-10-18 13-04-40](https://github.com/user-attachments/assets/67e2f026-8976-4b36-a292-07a3cb84001d)

Reply:
addressed in: [7616e23](https://github.com/redhat-ai-dev/ai-lab-template/pull/40/commits/7616e23c39f55d7dcdfaa0678c8a77c0784bb1b7)

Reply:
<img width="1217" alt="Screenshot 2024-10-18 at 4 19 28 PM" src="https://github.com/user-attachments/assets/2d7f4740-91de-4a6c-b833-2247659b9ec8">